### ISSUE TITLE
Background save shouldn't be triggered in splash screens

### ISSUE THREAD

**Issue by mal** [renpy/renpy#2045]
The `main_menu` var is used to prevent saving on the Main Menu, however this does not extend to any splash screens. For splash screens which have confirmations or other types of blocking interaction it's possible for game to be backgrounded on mobile devices and due to the `config.save_on_mobile_background` variable it will try to save.

This isn't really ideal, and in cases where the autosave name is constructed using game variables, can result in a crash as such vars are not defined during splash.

**tl;dr**: Save prevention should probably be extended to include splash screens.
